packages/xpetra/src/Map/Xpetra_TpetraMap_def.hpp

```diff
@@ -160,260 +156,157 @@ namespace Xpetra {
+{ XPETRA_MONITOR("TpetraMap::isSameAs"); return map_->isSameAs(toTpetra(map)); }
 
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+Teuchos::RCP< const Teuchos::Comm< int > >  TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::getComm() const
+{ XPETRA_MONITOR("TpetraMap::getComm"); return map_->getComm(); }
 #ifdef TPETRA_ENABLE_DEPRECATED_CODE
-    //! Get this Map's Node object.
-    Teuchos::RCP< Node >  getNode() const { XPETRA_MONITOR("TpetraMap::getNode"); return map_->getNode(); }
+
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+Teuchos::RCP< Node >  TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::getNode() const
+{ XPETRA_MONITOR("TpetraMap::getNode"); return map_->getNode(); }
 #endif // TPETRA_ENABLE_DEPRECATED_CODE
 
-    //@}
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+std::string TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::description() const
+{ XPETRA_MONITOR("TpetraMap::description"); return map_->description(); }
 
-    //! @name
-    //@{
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+void TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::describe(Teuchos::FancyOStream &out, const Teuchos::EVerbosityLevel verbLevel) const
+{ XPETRA_MONITOR("TpetraMap::describe"); map_->describe(out, verbLevel); }
 
-    //! Return a simple one-line description of this object.
-    std::string description() const { XPETRA_MONITOR("TpetraMap::description"); return map_->description(); }
-
-    //! Print this object with the given verbosity level to the given FancyOStream.
-    void describe(Teuchos::FancyOStream &out, const Teuchos::EVerbosityLevel verbLevel=Teuchos::Describable::verbLevel_default) const { XPETRA_MONITOR("TpetraMap::describe"); map_->describe(out, verbLevel); }
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::removeEmptyProcesses () const
+{
+    return toXpetra(map_->removeEmptyProcesses());
+}
 
-    RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > removeEmptyProcesses () const {
-      return toXpetra(map_->removeEmptyProcesses());
-    }
-    RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > replaceCommWithSubset (const Teuchos::RCP<const Teuchos::Comm<int> >& newComm) const {
-      return toXpetra(map_->replaceCommWithSubset(newComm));
-    }
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+RCP<const Map<LocalOrdinal, GlobalOrdinal, Node> > TpetraMap<LocalOrdinal,GlobalOrdinal,Node>::replaceCommWithSubset (const Teuchos::RCP<const Teuchos::Comm<int> >& newComm) const
+{
+    return toXpetra(map_->replaceCommWithSubset(newComm));
+}
 
 #ifdef XPETRA_ENABLE_DEPRECATED_CODE
-    template<class Node2>
-    RCP<Map<LocalOrdinal, GlobalOrdinal, Node2> > XPETRA_DEPRECATED clone(const RCP<Node2> &node2) const {
-      return toXpetraNonConst(map_->clone(node2));
-    }
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+template<class Node2>
```

Comment:
@csiefer2 `Node2` caused issues but I added the `template<class Node2>` entry and the compiler got happy with it.  Can you verify this is ok to do or am I breaking something subtle?

Reply:
That's correct.